### ISSUE TITLE
Design the remove user functionality

### ISSUE THREAD

**Issue by emil-muller** [cyber-crypt-com/encryptonize-core#63]
**Description**
For cleanliness and privilege revocation we want to remove users from the authstorage. There are however some complications and design decisions to be made before we can implement this functionality.

Some things we should consider:
- Should admins be able to remove admin users? And if so, how can we determine a users type?
- Should users be able to remove themselves? 
- Should it be possible to downgrade a user from admin to user?

**Enabler for**
List issues that have direct customer value that this story enables, if any
- Issue  cyber-crypt-com/tmp#62 

**Acceptance Criteria**
- A decision is agreed upon through a discussion with the development team
- The decision is uploaded as an ADR to our documentation repo

**Scope**
Design of the mentioned feature.

**Not in Scope**
The implementation of said feature.
